client/src/components/Donation/donation-modal.tsx

```diff
@@ -52,6 +53,30 @@ type DonateModalProps = {
   show: boolean;
 };
 
+const GetCommonDonationText = ({ ctaNumber }: { ctaNumber: number }) => {
+  const { t } = useTranslation();
+  // const useFeature;
+  const rotateProgressModalCta = useFeature('progress-modal-cta-rotation').on;
+  if (rotateProgressModalCta)
+    return <b>{t(`donate.progress-modal-cta-${ctaNumber}`)}</b>;
+
+  const donationDuration = modalDefaultDonation.donationDuration;
+  switch (donationDuration) {
+    case 'one-time':
+      return <b>{t('donate.duration')}</b>;
+    case 'month':
+      return <b>{t('donate.duration-2')}</b>;
+    default:
+      return <b>{t('donate.duration-4')}</b>;
+  }
+};
+
+function getctaNumberBetween1To10() {
+  const min = 1;
+  const max = 10;
```

Comment:
IMHO, I disagree. They're small enough values that the memory allocation is negligible, and they're values specific to this function - for maintainability, they should stay in this function.

Reply:
Whelp, this is going to be philosophy talk.

Just TD;LR, we are writing code for users not developers, we should account for how much GC is created for the user and how does that affect their experience.

This is a simple example, so it isn't worth it to discuss anyway, and there are multiples way for developer to get needed info quick, press <kbd>F12</kbd> or <kbd>Ctrl</kbd> to find definition, or you can use TypeScript and put the value as global and use `as const`, for this type of info.

![image](https://user-images.githubusercontent.com/88248797/236298891-7f80886b-f732-46a2-849b-d70fe4681fa8.png)

----

In general about small enough values that memory allocate, we don't know if that value will change in Node 20 or not, we don't know how many small values are there in the code base, and we even don't use React and use Element instead of JSX.

All of these are small, but they build up.

Reply:
Right, but your argument is that it should persist in memory for the lifetime of the component, rather than be instantiated when needed and cleaned up when done? That seems backwards to me. Am I misunderstanding?

Reply:
> rather than be instantiated when needed and cleaned up when done? That seems backwards to me. Am I misunderstanding?

What am I misunderstanding, for me the workflow is this:

- the camper finish the step.
- see the component
- the value is instantiated.
- close the component
- keep going on
- until model is showed again, but the value isn't instantiated again.


Why does the instantiated of the values every time the component is called a good thing?

Reply:
BTW, can we discuss this in chat or discord, I don't want to hold the PR for this.